iceoryx2-bb/container/src/vec.rs

```diff
@@ -136,6 +138,42 @@ impl<T> RelocatableContainer for Vec<T> {
     }
 }
 
+impl<T> Deref for Vec<T> {
+    type Target = [T];
+
+    fn deref(&self) -> &Self::Target {
+        self.verify_init(&format!("Vec<{}>::push()", std::any::type_name::<T>()));
+        unsafe { core::slice::from_raw_parts((*self.data_ptr.as_ptr()).as_ptr(), self.len) }
+    }
+}
+
+impl<T> DerefMut for Vec<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.verify_init(&format!("Vec<{}>::push()", std::any::type_name::<T>()));
+        unsafe {
+            core::slice::from_raw_parts_mut((*self.data_ptr.as_mut_ptr()).as_mut_ptr(), self.len)
+        }
+    }
+}
+
+impl<T: PartialEq> PartialEq for Vec<T> {
+    fn eq(&self, other: &Self) -> bool {
+        if other.len() != self.len() {
+            return false;
+        }
+
+        for i in 0..self.len() {
+            if other[i] != self[i] {
+                return false;
+            }
+        }
+
+        true
```

Comment:
Just as food for thought, this would be the equivalent approach with iterators
```rs
self.iter()
    .zip(other.iter())
    .try_for_each(|(lhs, rhs)| if lhs == rhs { Some(()) } else { None })
    .is_some()
```